### PR TITLE
[DBAAS-989] increment CRDs to v1beta1 apiVersion, change policy fields

### PR DESCRIPTION
**Pull request by tchughesiv**
Signed-off-by: Tommy Hughes <tohughes@redhat.com>

## Description
 - increment apiVersions to `v1beta1`
 - add conversion webhooks.
 - move and rename all connection policy fields to a new `connections` object.
   - `connectionNsSelector` -> `connections.nsSelector`
   - `connectionNamespaces` -> `connections.namespaces`
 - move all policy-related fields in `DBaaSInventories` to a new `spec.policy` object.
 - if ANYTHING is set in an inventory's `spec.policy`, even if it's set to `nil`... it will override everything in the dbaaspolicy object for that one inventory.
 - add coverage for `DBaaSPlatformReconciler`

related UI changes - https://github.com/RHEcosystemAppEng/dbaas-dynamic-plugin/pull/212

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA or in issue number have been completed
- [ ] Unit tests added that prove the fix is effective or the feature works 
- [ ] Documentation added for the feature
- [ ] all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

File: controllers/dbaasplatform_controller.go [hunk 1]
```diff
@@ -187,95 +180,28 @@ func (r *DBaaSPlatformReconciler) Reconcile(ctx context.Context, req ctrl.Reques
...
-	if err != nil {
-		return err
-	}
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&dbaasv1alpha1.DBaaSPlatform{}).
+		For(&v1beta1.DBaaSPlatform{}).
 		Complete(r)
 }
 
-func (r *DBaaSPlatformReconciler) createPlatformCR(ctx context.Context, serverClient k8sclient.Client) (*dbaasv1alpha1.DBaaSPlatform, error) {
```
Comment (redhatHameed) [inline controllers/dbaasplatform_controller.go:211]: any specific reason to move `createPlatformCR` into DBaaSDefaultPolicyReconciler ?

File: bundle/manifests/dbaas.redhat.com_dbaasinventories.yaml [hunk 2]
```diff
@@ -6,6 +6,17 @@ metadata:
   creationTimestamp: null
   name: dbaasinventories.dbaas.redhat.com
 spec:
+  conversion:
+    strategy: Webhook
+    webhook:
+      clientConfig:
+        service:
+          name: dbaas-operator-webhook-service
+          namespace: openshift-dbaas-operator
```
Comment (redhatHameed) [inline bundle/manifests/dbaas.redhat.com_dbaasinventories.yaml:15]: we have different namespace name in production, does that will make any difference/cause or any issue with webhook ?

File: bundle/manifests/dbaas-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -606,7 +794,23 @@ spec:
...
-    - v1
+    - v1alpha1
+    - v1beta1
+    containerPort: 443
+    conversionCRDs:
+    - dbaasconnections.dbaas.redhat.com
+    - dbaasinstances.dbaas.redhat.com
+    - dbaasinventories.dbaas.redhat.com
+    - dbaaspolicies.dbaas.redhat.com
+    - dbaasproviders.dbaas.redhat.com
```
Comment (redhatHameed) [inline bundle/manifests/dbaas-operator.clusterserviceversion.yaml:805]: do we need to add for `dbaasplatforms.dbaas.redhat.com` or not required ?

File: bundle/manifests/dbaas.redhat.com_dbaasinventories.yaml (hunk 2 above)
Comment (tchughesiv) [reply bundle/manifests/dbaas.redhat.com_dbaasinventories.yaml:15]: no issue, OLM is supposed to reconfigure these settings dynamically based on the install namespace.

File: controllers/dbaasplatform_controller.go (hunk 1 above)
Comment (tchughesiv) [reply controllers/dbaasplatform_controller.go:211]: this was done to ensure platform installs would succeed... before I realized there was a bug in OLM - https://issues.redhat.com/browse/OCPBUGS-4991.

It's my hope/expectation that, once this is resolved, I can safely revert this particular change.

File: bundle/manifests/dbaas-operator.clusterserviceversion.yaml (hunk 3 above)
Comment (tchughesiv) [reply bundle/manifests/dbaas-operator.clusterserviceversion.yaml:805]: Technically not required... BUT...
this was done to ensure platform installs would succeed... before I realized there was a bug in OLM - https://issues.redhat.com/browse/OCPBUGS-4991.

It's my expectation that, once this is resolved, we can also increment the dbaasplatform CRD. That said, it's worth discussing whether it's even necessary considering this operator is the only consumer.

**Comment by jianrongzhang89** [comment]
@tchughesiv could you add the steps of test & verification in the PR description?

**Review by redhatHameed — APPROVED** [review]

File: bundle/manifests/dbaas-operator.clusterserviceversion.yaml (hunk 3 above)
Comment (tchughesiv) [reply bundle/manifests/dbaas-operator.clusterserviceversion.yaml:805]: working on this here - https://github.com/RHEcosystemAppEng/dbaas-operator/pull/306

File: controllers/dbaasplatform_controller.go (hunk 1 above)
Comment (tchughesiv) [reply controllers/dbaasplatform_controller.go:224]: working on this here - https://github.com/RHEcosystemAppEng/dbaas-operator/pull/306

Comment (tchughesiv) [reply controllers/dbaasplatform_controller.go:224]: some of the errors i'm seeing trying to revert to the old method. we may have to keep these changes if i can't find a way forward - 
```
2023-01-13T16:47:13.248Z INFO controller-runtime.builder Conversion webhook enabled {"GVK": "dbaas.redhat.com/v1beta1, Kind=DBaaSProvider"}
2023-01-13T16:47:16.305Z ERROR setup unable to create controller {"controller": "DBaaSPlatform", "error": "could not get a list of dbaas platform intallation CR: conversion webhook for dbaas.redhat.com/v1alpha1, Kind=DBaaSPlatform failed: the server could not find the requested resource"}
main.main
/opt/app-root/src/main.go:211
runtime.main
/usr/lib/golang/src/runtime/proc.go:255
```

**Pull request merged**